From 263c1b1ee1a195092326055dfe48b1663a820077 Mon Sep 17 00:00:00 2001
From: Paul Cercueil <paul@crapouillou.net>
Date: Wed, 10 Jan 2018 22:21:12 +0100
Subject: [PATCH] CMakeLists.txt: Don't require ASM compiler

Signed-off-by: Paul Cercueil <paul@crapouillou.net>
---
 CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 48c6fa287..2098bd208 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -31,7 +31,7 @@ endif(POLICY CMP0054)
 
 set(CMAKE_CONFIGURATION_TYPES "Release;Debug;MinSizeRel;RelWithDebInfo")
 
-project(libpng ASM C)
+project(libpng C ASM)
 enable_testing()
 
 set(PNGLIB_MAJOR 1)
-- 
2.11.0

