From 7fa5789343336d155de4f0da91ff3497a07a8872 Mon Sep 17 00:00:00 2001
From: Paul Cercueil <paul@crapouillou.net>
Date: Fri, 19 Jan 2018 16:37:55 +0100
Subject: [PATCH] sigc++: Fix CMake error on Windows

Signed-off-by: Paul Cercueil <paul@crapouillou.net>
---
 CMakeLists.txt        |  6 ++++--
 sigc++/CMakeLists.txt | 13 +++++++++++--
 2 files changed, 15 insertions(+), 4 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index d8d56b2..639dbbd 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -71,8 +71,10 @@ enable_testing()
 
 
 add_subdirectory (sigc++)
-add_subdirectory (examples)
-add_subdirectory (tests)
+if (NOT MSVC)
+	add_subdirectory (examples)
+	add_subdirectory (tests)
+endif (NOT MSVC)
 
 
 set (PROJECT_CMAKE_NAME		"${PROJECT_NAME}-3")
diff --git a/sigc++/CMakeLists.txt b/sigc++/CMakeLists.txt
index c06d829..b53ea3e 100644
--- a/sigc++/CMakeLists.txt
+++ b/sigc++/CMakeLists.txt
@@ -21,12 +21,19 @@ set (SOURCE_FILES
 	functors/slot_base.cc
 )
 
+add_definitions(-DSIGC_BUILD=1)
+
 set (SIGCPP_LIB_NAME sigc-${SIGCXX_API_VERSION})
 
 add_library(${SIGCPP_LIB_NAME} SHARED ${SOURCE_FILES})
 
-set_property (TARGET ${SIGCPP_LIB_NAME} PROPERTY VERSION ${PACKAGE_VERSION})
-set_property(TARGET ${SIGCPP_LIB_NAME}  PROPERTY SOVERSION ${LIBSIGCPP_SOVERSION})
+set_target_properties(${SIGCPP_LIB_NAME} PROPERTIES
+		VERSION ${PACKAGE_VERSION}
+		SOVERSION ${LIBSIGCPP_SOVERSION}
+		CXX_STANDARD 11
+		CXX_STANDARD_REQUIRED ON
+		CXX_EXTENSIONS OFF
+)
 
 set (INCLUDE_INSTALL_DIR "include/${PROJECT_NAME}-${SIGCXX_API_VERSION}")
 
@@ -42,5 +49,7 @@ install (
     TARGETS ${SIGCPP_LIB_NAME}
     EXPORT "${PROJECT_CMAKE_NAME}Targets"
     LIBRARY DESTINATION "lib"
+    ARCHIVE DESTINATION "lib"
+    RUNTIME DESTINATION "bin"
     INCLUDES DESTINATION "${INCLUDE_INSTALL_DIR}"
 )
-- 
2.11.0

